src/Jonathan_Requirements/Jonathan_Testbench.c

```diff
@@ -0,0 +1,62 @@
+#include "Jonathan_Requirements10_11_12_19.h"
+#include "Jonathan_Requirement9.h"
+#include <stdio.h>
+#include <time.h>
+#include <stdlib.h>
+#include <GL/glew.h>
+
+///Ref: Req. 10,11,12,19
+/* 
+ Generates a random integer for movement speed to test draw distance per requirements 11 & 12.
+ Uses current weather conditions in draw distance calculation per requirement 10.
+ */
+void test_draw_speed(Precipitation *P)
+{
+    float speed = rand() % 20;
+    printf("speed is: %f\n", speed);
+    int x = 0;
+    int *render_radius = &x;
+
+    *render_radius = set_draw_distance(render_radius, speed, P);
+    printf("render_radius is: %d\n", *render_radius);
+}
+///Ref: Req. 19
+/*
+Generates random vectors in degrees of 360 or less for precipitation directions
+*/
+void test_precip_vectors(Precipitation *P)
+{
+    float x = rand() % 360;
+    float y = rand() % 360;
+    float z = rand() % 360;
+    generate_precipitation_vectors(P, x, y, z);
+    printf("precip vectors: x: %f, y: %f, z: %f\n", P->pv.x, P->pv.y, P->pv.z);
+}
+
+///Ref: Req. 19
+/*
+Generates a random number from 0 to max number of weather conditions and sets the enumerated precipitation type to the random
+number to test the precipitation functionality with movement speed and draw distance.
+*/
+void test_precipitation(Precipitation *P)
+{
+    int i = 0;
+    while (i <= CLEAR) //loop is here instead of hard-coded int incase more precip types added later
+    {
+        i++;
+    }
+    int precip_type = rand() % i;
+    printf("precip type is: %d\n", precip_type);
+    set_precipitation_type(P, precip_type);
+}
+
+int main(int argc, char **argv)
```

Comment:
The precipitation vectors look good. Is there currently a visual to indicate the change in weather, or does it currently just affect the view distance?

Also, for your tests, what is being asserted for each test?

Reply:
Right now it just affects the view distance depending on which weather condition is selected. The vectors are there for development of actual weather changes that I'm not sure we will be able to fully implement at this point. I think we might actuallly want the vectors in radians but that's an easy change if we determine it's needed.

The tests I initally hard codded values to ensure edge cases were functional then I threw in a random number generator to just select weather conditions at random on each run. I don't have anything that definitely states "pass" or "fail". I was mostly just looking to ensure the code in place was operational with print statements to track progress and help determine as new changes were made if they broke the system or not.